app/policies/subject_rule_effect_policy.rb

```diff
@@ -38,26 +21,38 @@ def edit?
   end
 
   def update?
-    # record is a subject_rule_effect from the controller
-    if credential.admin?
-      true
-    else
-      # this is a good place to check the defined subect set id in the
-      # rule effect belongs to the scoped credential project ids using the API client
+    return true if credential.admin?
+    return false unless valid_credentials?
+
+    if record.config.key?('subject_set_id')
```

Comment:
Not blocking - ideally the record would tell us what kind of effect it is vs having to dig into the data structure to see what key there is. I think you can do this to check (this was not at all obvious to me, i had to dig into the code to see how this worked)
```
record.effect.is_a?(Effects::AddSubjectToSet)
```

Reply:
Nice. I like that.